src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Unix.cs

```diff
@@ -90,18 +102,54 @@ public ClientCertificateOption ClientCertificateOptions
                 }
                 else
                 {
-                    _managedHandler.ClientCertificateOptions = value;
+                    switch (value)
+                    {
+                        case ClientCertificateOption.Manual:
+                            ThrowForModifiedManagedSslOptionsIfStarted();
+                            _managedHandler.SslOptions.LocalCertificateSelectionCallback = null;
+                            break;
+
+                        case ClientCertificateOption.Automatic:
+                            ThrowForModifiedManagedSslOptionsIfStarted();
+                            _managedHandler.SslOptions.LocalCertificateSelectionCallback = (sender, targetHost, localCertificates, remoteCertificate, acceptableIssuers) => CertificateHelper.GetEligibleClientCertificate();
```

Comment:
@bartonjs, could you help me validate that this makes sense?  It appears to be the same thing WinHttpHandler is doing.

Reply:
Assuming that the SslStream object isn't exposed, `localCertificates` should always be empty (an insight I didn't have this morning before coffee).  It would, theoretically, be more appropriate to filter the certificates list by the `acceptableIssuers` value, but since the trend in TLS server configurations these days is to not send the list (because it's big and no one pays attention to it) maybe it doesn't matter.

Likewise, it's hard for the automatic mode to really do anything based on what the server cert/chain are... so... I guess... this is a right answer.

Since this is what NetFx/CoreFx-Windows is doing, it seems fine to me.  Definitely nonintuitive :)

You'll want to double check that this code, on Linux, doesn't use OpenExistingOnly when opining the CU\My store.  Windows account creation provisions the CU\My store, so it exists for users (except those who went out of their way to delete it).  X509Store is our own thing on Linux, with no auto-provision.  So OpenExistingOnly will cause a "never added any certs to CU\My" user to get an exception.

Reply:
> so... I guess... this is a right answer

:smile:

> You'll want to double check that this code, on Linux, doesn't use OpenExistingOnly when opining the CU\My store

Ah, ok.  Right now it's:
```C#
myStore.Open(OpenFlags.ReadOnly);
```
On Unix it should be:
```C#
myStore.Open(OpenFlags.OpenExistingOnly | OpenFlags.ReadOnly);
```
?

Reply:
Correct, Unix should just set OpenFlags.ReadOnly.